heft-plugins/heft-storybook-plugin/src/StorybookPlugin.ts

```diff
@@ -67,135 +71,118 @@ export interface IStorybookPluginOptions {
     if (parseResult.error) {
       throw new Error(
-        `The ${TASK_NAME} task cannot start because the "storykitPackageName"` +
+        `The ${taskSession.taskName} task cannot start because the "storykitPackageName"` +
           ` plugin option is not a valid package name: ` +
           parseResult.error
       );
     }
-    this._storykitPackageName = options.storykitPackageName;
 
-    if (!options.startupModulePath) {
-      throw new Error(
-        `The ${TASK_NAME} task cannot start because the "startupModulePath"` +
-          ` plugin option was not specified`
-      );
-    }
-    this._startupModulePath = options.startupModulePath;
-
-    const storybookParameters: IHeftFlagParameter = heftSession.commandLine.registerFlagParameter({
-      associatedActionNames: ['start'],
-      parameterLongName: '--storybook',
-      description:
-        '(EXPERIMENTAL) Used by the "@rushstack/heft-storybook-plugin" package to launch Storybook.'
-    });
-
-    heftSession.hooks.build.tap(PLUGIN_NAME, (build: IBuildStageContext) => {
-      if (!storybookParameters.actionAssociated || !storybookParameters.value) {
-        this._logger.terminal.writeVerboseLine(
-          'The command line does not include "--storybook", so bundling will proceed without Storybook'
+    // Only tap if the --storybook flag is present.
+    if (storybookParameter.value) {
+      const configureWebpackTapOptions: { name: string, stage: number } = {
+        name: PLUGIN_NAME,
+        stage: Number.MAX_SAFE_INTEGER
+      };
+      const configureWebpackTap: () => Promise<null> = async () => {
+        // Discard Webpack's configuration to prevent Webpack from running
+        this._logger.terminal.writeLine(
+          'The command line includes "--storybook", redirecting Webpack to Storybook'
         );
-        return;
-      }
-
-      this._logger.terminal.writeVerboseLine(
-        'The command line includes "--storybook", redirecting Webpack to Storybook'
+        return null;
+      };
+
+      taskSession.requestAccessToPluginByName(
+        '@rushstack/heft-webpack4-plugin',
+        WEBPACK_PLUGIN_NAME,
+        (accessor: IWebpack4PluginAccessor) =>
+          accessor.onConfigureWebpackHook.tapPromise(configureWebpackTapOptions, configureWebpackTap)
       );
 
-      build.hooks.preCompile.tap(PLUGIN_NAME, (preCompile: IPreCompileSubstage) => {
-        preCompile.hooks.run.tapPromise(PLUGIN_NAME, () => {
-          return this._onPreCompileAsync(heftConfiguration);
-        });
-      });
-
-      build.hooks.bundle.tap(PLUGIN_NAME, (bundle: IBundleSubstage) => {
-        bundle.hooks.configureWebpack.tap(
-          { name: PLUGIN_NAME, stage: Number.MAX_SAFE_INTEGER },
-          (webpackConfiguration: unknown) => {
-            // Discard Webpack's configuration to prevent Webpack from running
-            return null;
-          }
-        );
+      taskSession.requestAccessToPluginByName(
+        '@rushstack/heft-webpack5-plugin',
```

Comment:
Put these in typechecked consts?

Reply:
I'm going to leave this for now. If the package name were to change, it would also facilitate a change to package.json, so at that point it'd have to be pretty explicit. Importing from the package means you're already updating the import that the type comes from. Seems to have more value for the plugin name (which can be changed by the plugin dev) than the package name (which already requires a change on consumer side anyway).